tests/test_metadata.py

```diff
@@ -31,8 +32,129 @@ def test_select_lines():
 
 """
 
-class HandleConfigVersionTests(unittest.TestCase):
+@contextmanager
+def tmp_meta(spec, index=None):
+    import os.path
+    import shutil
+    import tempfile
+    import textwrap
+
+    recipe_dir = tempfile.mkdtemp(prefix='tmp_recipe_')
+    try:
+        with open(os.path.join(recipe_dir, 'meta.yaml'), 'w') as fh:
+            fh.write(textwrap.dedent(spec))
+        m = MetaData(recipe_dir)
+        if index is not None:
+            m._index = index
+        yield m
+    finally:
+        shutil.rmtree(recipe_dir)
+
+
+class TestResolvedDepends(unittest.TestCase):
```

Comment:
Think I may want to add a test which pins something which wasn't a build dependency.